### ISSUE TITLE
`paymentItems` is not reactive

### ISSUE THREAD

**Issue by dickermoshe** [google-pay/flutter-plugin#235]
## BUG

If you change the value of `paymentItems` on a PayButton it doesn't update it internally.

## Example 
This Widget returns a ApplePayButton that will charge `priceInCents`.
However, If the parent widget changes the price (E.G. The customer selects a different tip amount) the payment will still show the original amount.

```dart
class PaymentButton extends HookConsumerWidget {
  const PaymentButton(
      {super.key,
      required this.priceInCents,
      this.onPressed,
      this.onSuccess,
      this.onError});
  final int priceInCents;
  final VoidCallback? onPressed;
  final VoidCallback? onSuccess;
  final void Function(Object?)? onError;

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final paymentItems = [
      p.PaymentItem(
          label: "Donation",
          amount: (priceInCents / 100).toStringAsFixed(2),
          type: p.PaymentItemType.total,
          status: p.PaymentItemStatus.final_price)
    ];

    return p.ApplePayButton(
      onPressed: onPressed,
      paymentConfiguration:
          p.PaymentConfiguration.fromJsonString(defaultApplePay),
      type: p.ApplePayButtonType.donate,
      onPaymentResult: (result) {
        runApplePay(result, amountInCents: priceInCents);
        if (onSuccess != null) onSuccess!();
      },
      paymentItems: paymentItems,
    );
  }
}
```

https://github.com/google-pay/flutter-plugin/assets/75931499/ee77d93a-1026-4060-a001-4c429475a82b

## Temp Workaround
Assign the button a `UniqueKey` and change it when the `priceInCents` changes.
This will force Flutter to rebuild the entire `ApplePayButton`
(This is using flutter_hooks, but you could use something else to do this.)
```dart
 final buttonKey = useMemoized(() => UniqueKey(), [priceInCents]);
```

**Comment by dickermoshe** [comment]
This is because `onMethodCallHandler` is only called once.
https://github.com/google-pay/flutter-plugin/blob/af2e70d121778337bf4bda9b5ef12ffbe4df664d/pay_ios/lib/src/widgets/apple_pay_button.dart#L153-L162

**Comment by dickermoshe** [comment]
PR https://github.com/google-pay/flutter-plugin/pull/236 fixes this.
Please upvote

**Comment by JlUgia** [comment]
Published under [`pay_ios:1.0.11`](https://pub.dev/packages/pay_ios/versions/1.0.11) and [`pay:1.2.0-beta02`](https://pub.dev/packages/pay/versions/1.2.0-beta02).

**Comment by xOldeVx** [comment]
@dickermoshe what about Android, same issue there 👀
